deps/spidershim/include/v8.h

```diff
@@ -1149,18 +1152,85 @@ class V8_EXPORT String : public Name {
   bool IsExternal() const { return false; }
   bool IsExternalOneByte() const { return false; }
 
-  class V8_EXPORT ExternalOneByteStringResource {
+  class V8_EXPORT ExternalStringResourceBase {  // NOLINT
    public:
-    virtual ~ExternalOneByteStringResource() {}
-    virtual const char *data() const = 0;
-    virtual size_t length() const = 0;
+    virtual ~ExternalStringResourceBase() {}
+
+    virtual bool IsCompressible() const { return false; }
+
+   protected:
+    ExternalStringResourceBase() {}
+
+    /**
+     * Internally V8 will call this Dispose method when the external string
+     * resource is no longer needed. The default implementation will use the
+     * delete operator. This method can be overridden in subclasses to
+     * control how allocated external string resources are disposed.
+     */
+    virtual void Dispose() { delete this; }
+
+   private:
+    // Disallow copying and assigning.
+    ExternalStringResourceBase(const ExternalStringResourceBase&);
+    void operator=(const ExternalStringResourceBase&);
+
+    friend class v8::internal::Heap;
```

Comment:
Since we have our own `v8::internal` namespace, please delete this friend, and also the forward declaration of the `Heap` class in the beginning of `v8.h`.